classes/ForbiddenSeal.php

```diff
@@ -1,336 +1,398 @@
-        $this->avatar_size = self::$benefits[$this->level]['avatar_size'];
-        $this->avatar_filesize = self::$benefits[$this->level]['avatar_filesize'];
-        $this->avatar_styles = self::$benefits[$this->level]['avatar_styles'];
-        $this->logout_timer = self::$benefits[$this->level]['logout_timer'];
-        $this->inbox_size = self::$benefits[$this->level]['inbox_size'];
-        $this->journal_size = self::$benefits[$this->level]['journal_size'];
-        $this->journal_image_x = self::$benefits[$this->level]['journal_image_x'];
-        $this->journal_image_y = self::$benefits[$this->level]['journal_image_y'];
-        $this->journal_youtube_embed = self::$benefits[$this->level]['journal_youtube_embed'];
-        $this->chat_post_size = self::$benefits[$this->level]['chat_post_size'];
-        $this->pm_size = self::$benefits[$this->level]['pm_size'];
-        $this->extra_jutsu_equips = self::$benefits[$this->level]['extra_jutsu_equips'];
-        $this->extra_weapon_equips = self::$benefits[$this->level]['extra_weapon_equips'];
-        $this->extra_armor_equips = self::$benefits[$this->level]['extra_armor_equips'];
-        $this->long_training_time = self::$benefits[$this->level]['long_training_time'];
-        $this->long_training_gains = self::$benefits[$this->level]['long_training_gains'];
-        $this->extended_training_time = self::$benefits[$this->level]['extended_training_time'];
-        $this->extended_training_gains = self::$benefits[$this->level]['extended_training_gains'];
-        $this->stat_transfer_boost = self::$benefits[$this->level]['stat_transfer_boost'];
-        $this->extra_stat_transfer_points_per_ak = self::$benefits[$this->level]['extra_stat_transfer_points_per_ak'];
-        $this->max_battle_history_view = self::$benefits[$this->level]['max_battle_history_view'];
-        $this->free_transfer_bonus = self::$benefits[$this->level]['free_transfer_bonus'];
-        $this->bonus_pve_reputation = self::$benefits[$this->level]['bonus_pve_reputation'];
-
-        // Display variables
-        $this->name_color_display = self::$benefits[$this->level]['name_color_display'];
-        $this->journal_image_display = self::$benefits[$this->level]['journal_image_display'];
-        $this->avatar_size_display = self::$benefits[$this->level]['avatar_size_display'];
+    public static function getNameColorDisplay(array $name_colors): string {
+        $return = '';
+        $count = 0;
+        foreach($name_colors as $name => $className) {
+            if($name == 'default') {
+                continue;
+            }
+            $return .= "<span class='{$className}' style='font-weight:bold;'>" . ucwords($name) . "</span>";
+            $count++;
+            if($count%2 == 0) {
+                $return .= "<br /> ";
+            }
+            else {
+                $return .= '/';
+            }
+        }
+        return substr($return, 0, strlen($return)-1);
+    }
```

Comment:
Having display calculated in a class function is a pattern I think we should move away from - Could you move this to the template where it's used? Okay to make it a helper function inside the template